armi/reactor/tests/test_parameters.py

```diff
@@ -509,228 +523,80 @@ class MockSyncPC(parameters.ParameterCollection):
-    system.
-    """
+class SynchronizationTests(unittest.TestCase):
+    """Some tests that must be run with mpirun instead of the standard unittest system."""
 
     def setUp(self):
         self.r = makeComp("reactor")
         self.r.core = makeComp("core")
         self.r.add(self.r.core)
-        for ai in range(context.MPI_SIZE * 4):
+        for ai in range(context.MPI_SIZE * 3):
             a = makeComp("assembly{}".format(ai))
             self.r.core.add(a)
-            for bi in range(10):
+            for bi in range(3):
                 a.add(makeComp("block{}-{}".format(ai, bi)))
+
         self.comps = [self.r.core] + self.r.core.getChildren(deep=True)
         for pd in MockSyncPC().paramDefs:
             pd.assigned = parameters.NEVER
 
-    def tearDown(self):
-        del self.r
-
-    def run(self, testNamePrefix="mpitest_"):
-        with open("mpitest{}.temp".format(context.MPI_RANK), "w") as self.l:
-            for methodName in sorted(dir(self)):
-                if methodName.startswith(testNamePrefix):
-                    self.write("{}.{}".format(self.__class__.__name__, methodName))
-                    try:
-                        self.setUp()
-                        getattr(self, methodName)()
-                    except Exception:
-                        self.write("failed, big time")
-                        traceback.print_exc(file=self.l)
-                        self.write("*** printed exception")
-                        try:
-                            self.tearDown()
-                        except:  # noqa: bare-except
-                            pass
-
-            self.l.write("done.")
-
-    def write(self, msg):
-        self.l.write("{}\n".format(msg))
-        self.l.flush()
-
-    def assertRaises(self, exceptionType):
-        class ExceptionCatcher:
-            def __enter__(self):
-                pass
-
-            def __exit__(self, exc_type, exc_value, traceback):
-                if exc_type is exceptionType:
-                    return True
-                raise AssertionError(
-                    "Expected {}, but got {}".format(exceptionType, exc_type)
-                )
-
-        return ExceptionCatcher()
+    @unittest.skipIf(context.MPI_SIZE <= 1 or MPI_EXE is None, "Parallel test only")
+    def test_noConflicts(self):
+        """Make sure sync works across processes.
 
-    def assertEqual(self, expected, actual):
-        if expected != actual:
-            raise AssertionError(
-                "(expected) {} != {} (actual)".format(expected, actual)
-            )
-
-    def assertNotEqual(self, expected, actual):
-        if expected == actual:
-            raise AssertionError(
-                "(expected) {} == {} (actual)".format(expected, actual)
-            )
+        .. test:: Synchronize a reactor's state across processes.
+            :id: T_ARMI_CMP_MPI0
+            :tests: R_ARMI_CMP_MPI
+        """
+        syncCount = self.r.syncMpiState()
```

Comment:
I think this is unneeded? Or at least the returned `syncCount` is unused, right? If so, either remove this line or replace `syncCount` with `_`, perhaps.

Reply:
I think something must have changed in practice in the last 4 or 5 years since these tests were written.  It appears we sync the MPI state more often, so now the only way to flip all the parameter sync flags to their correct starting positions is to call a `syncMpiState()` while everything is still clean.  (Or replicate a ton of work with things like `ps.assigned = parameters.NEVER`, as you commented on above.